src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -12,8 +12,7 @@ smt2_incremental_decision_proceduret::smt2_incremental_decision_proceduret(
 
 exprt smt2_incremental_decision_proceduret::handle(const exprt &expr)
 {
-  UNIMPLEMENTED;
```

Comment:
I do not understand this change and its commit message.

Why was this changed from `UNIMPLEMENTED` to just returning what was passed (it also loses the `const` modifier, unless I'm mistaken?)

Reply:
The `const` modifier is not "lost", as the type of this function is the same before and after this commit. The parameter is passed in by const reference and a copy is returned by value, therefore this is const correct and the return value cannot be used to mutate any other version of the expression.

Without this commit the console output from the test case is as follows, (both with and without `--slice-formula`)
```
Reason: Unimplemented handle of:
  constant
    * type: bool
    * value: true
```
This doesn't give us enough information in the regression test, to confirm that `--slice-formula` removes the expressions from `cprover_initialise` which we don't plan to support in the first slice of work. Implementing as the identity/copy function enables the test case to run a little further before hitting the unimplemented `set_to` member function which gives us more useful diagnostics.

There is likely a better way to implement this function. Unfortunately it is not documented in the base class. You can see the other current implementations of it here - https://github.com/diffblue/cbmc/blob/develop/src/solvers/smt2/smt2_conv.cpp#L756 and here - https://github.com/diffblue/cbmc/blob/develop/src/solvers/prop/prop_conv_solver.cpp#L40